### PR TITLE
MessageProcessor base implementation

### PR DESCRIPTION
**Pull request by i-Alex**

### CODE REVIEW

File: sdk/src/main/java/com/horizen/account/state/MessageProcessor.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.horizen.account.state;
+
+public interface MessageProcessor {
+    // Initialization is going to happen only once at genesis State creation.
+    // Common pattern: declare a new fake smart contract account in the View
+    void init(AccountStateView view) throws MessageProcessorInitializationException;
```
Comment (alsala) [inline sdk/src/main/java/com/horizen/account/state/MessageProcessor.java:6]: It is not clear to me why we need the view as input parameter. For instance we will use this init() call to setup the fake smart contract address which is not dependent on the view

Comment (i-Alex) [reply sdk/src/main/java/com/horizen/account/state/MessageProcessor.java:6]: In this example we should store the new Account in the view.
So the can be modified during the initialization.

File: sdk/src/main/java/com/horizen/account/state/ExecutionSucceeded.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package com.horizen.account.state;
+
+import java.math.BigInteger;
+
+
+// Message was executed as expected.
+public class ExecutionSucceeded implements ExecutionResult {
+    private final BigInteger gasUsed; // Total used gas during Message processing
+    private final byte[] returnData; // Returned data from invocation(function result or data supplied with revert opcode)
```
Comment (JcBernack) [inline sdk/src/main/java/com/horizen/account/state/ExecutionSucceeded.java:9]: Where do we put the logs? Also here?

File: sdk/src/main/java/com/horizen/account/state/MessageProcessor.java (hunk 1 above)
Comment (JcBernack) [reply sdk/src/main/java/com/horizen/account/state/MessageProcessor.java:6]: This is the place where e.g. the ForgerStakesProcessor can make sure that its account exists and is not declared as "empty" by setting some (arbitrary) code hash, right?

Comment (i-Alex) [reply sdk/src/main/java/com/horizen/account/state/MessageProcessor.java:6]: right

File: sdk/src/main/java/com/horizen/account/state/ExecutionSucceeded.java (hunk 2 above)
Comment (i-Alex) [reply sdk/src/main/java/com/horizen/account/state/ExecutionSucceeded.java:9]: I think, that logs are add to the View directly while processing message. Similar to the Geth implementation.
So, inside StateDB we cache the logs linked to tx id.

**Review by alsala — APPROVED** [review]

**Review by saratnt — APPROVED** [review]

**Review by JcBernack — APPROVED** [review]

**Pull request merged**